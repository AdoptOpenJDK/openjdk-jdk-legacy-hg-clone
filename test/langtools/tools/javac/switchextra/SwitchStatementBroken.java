/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Verify that rule and ordinary cases cannot be mixed.
 * @compile/fail/ref=SwitchStatementBroken.out -XDrawDiagnostics --enable-preview -source 13 SwitchStatementBroken.java
 */

public class SwitchStatementBroken {

    private void statementBroken(int i) {
        String res;

        switch (i) {
            case 0 -> { res = "NULL-A"; }
            case 1: { res = "NULL-A"; break; }
            case 2: { res = "NULL-A"; break; }
            default -> { res = "NULL-A"; break; }
        }
    }

}
