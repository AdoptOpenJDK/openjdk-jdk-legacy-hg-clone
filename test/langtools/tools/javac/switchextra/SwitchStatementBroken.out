SwitchStatementBroken.java:15:13: compiler.err.switch.mixing.case.types
- compiler.note.preview.filename: SwitchStatementBroken.java
- compiler.note.preview.recompile
1 error
